core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/insert.kt

```diff
@@ -10,53 +10,230 @@ import org.jetbrains.kotlinx.dataframe.annotations.Interpretable
 import org.jetbrains.kotlinx.dataframe.util.DEPRECATED_ACCESS_API
+import org.jetbrains.kotlinx.dataframe.util.INSERT_AFTER_COL_PATH
+import org.jetbrains.kotlinx.dataframe.util.INSERT_AFTER_COL_PATH_REPLACE
 import kotlin.reflect.KProperty
 
 // region DataFrame
 
 // region insert
 
+/**
+ * This function does not immediately insert the new column but instead specify a column to insert and
+ * returns an [InsertClause],
+ * which serves as an intermediate step.
+ * The [InsertClause] object provides methods to insert a new column using:
+ * - [under][InsertClause.under] - inserts a new column under the specified column group.
+ * - [after][InsertClause.after] - inserts a new column after the specified column.
+ * - [at][InsertClause.at]- inserts a new column at the specified position.
+ *
+ * Each method returns a new [DataFrame] with the inserted column.
+ *
+ * Check out [Grammar].
+ *
+ * @include [SelectingColumns.ColumnGroupsAndNestedColumnsMention]
+ *
+ * See [Selecting Columns][InsertSelectingOptions].
+ *
+ * For more information: {@include [DocumentationUrls.Insert]}
+ *
+ * See also:
+ * - [move][DataFrame.move] - move columns to a new position within the [DataFrame].
+ * - [add][DataFrame.add] - add new columns to the [DataFrame]
+ * (without specifying a position, to the end of the [DataFrame]).
+ */
+internal interface InsertDocs {
+
+    /**
+     * {@comment Version of [SelectingColumns] with correctly filled in examples}
+     * @include [SelectingColumns] {@include [SetInsertOperationArg]}
+     */
+    interface InsertSelectingOptions
+
+    /**
+     * ## Insert Operation Grammar
+     * {@include [LineBreak]}
+     * {@include [DslGrammarLink]}
+     * {@include [LineBreak]}
+     *
+     * **[`insert`][insert]**`(column: `[`DataColumn`][DataColumn]`)`
+     *
+     * {@include [Indent]}
+     * `| `[`insert`][insert]`(name: `[`String`][String]`, infer: `[`Infer`][Infer]`, rowExpression: `[`RowExpression`][RowExpression]`  }`
+     *
+     * {@include [Indent]}
+     * __`.`__[**`under`**][InsertClause.under]`  {  `**`column: `[`ColumnSelector`][ColumnSelector]**` }`
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`under`**][InsertClause.under]`(columnPath: `[`ColumnPath`][ColumnPath]`)`
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`after`**][InsertClause.after]`  {  `**`column: `[`ColumnSelector`][ColumnSelector]**` }`
+     *
+     * {@include [Indent]}
+     * `| `__`.`__[**`at`**][InsertClause.at]`(position: `[`Int`][Int]`)`
+     */
+    interface Grammar
+}
+
+/** {@set [SelectingColumns.OPERATION] [insert][insert]} */
+@ExcludeFromSources
+private interface SetInsertOperationArg
+
+/**
+ * Inserts the given [column] into this [DataFrame].
+ *
+ * {@include [InsertDocs]}
+ *
+ * ### Examples:
+ * ```kotlin
+ * // Insert a new column "age" under the column group with path ("info", "personal").
+ * df.insert(age).under(pathOf("info", "personal"))
```

Comment:
I'd love the examples, the main problem as usual - the update of these examples, but if we are close to the 1.0 - let's practice with it